src/pybind/mgr/dashboard_v2/controllers/rbd.py

```diff
@@ -68,3 +95,35 @@ def get(self, pool_name):
         if status == ViewCache.VALUE_EXCEPTION:
             raise value
         return {'status': status, 'value': value}
+
+    def create(self, data):
+        if not self.rbd:
+            self.rbd = rbd.RBD()
+
+        # Get input values
+        name = data.get('name')
+        pool_name = data.get('pool_name')
+        size = data.get('size')
+        obj_size = data.get('obj_size')
+        features = data.get('features')
+        stripe_unit = data.get('stripe_unit')
+        stripe_count = data.get('stripe_count')
+        data_pool = data.get('data_pool')
+
+        # Set order
+        order = None
+        if obj_size and obj_size > 0:
+            order = int(round(math.log(float(obj_size), 2)))
+
+        # Set features
+        feature_bitmask = self._format_features(features)
+
+        ioctx = mgr.rados.open_ioctx(pool_name)
+
+        try:
+            self.rbd.create(ioctx, name, size, order=order, old_format=False,
+                            features=feature_bitmask, stripe_unit=stripe_unit,
+                            stripe_count=stripe_count, data_pool=data_pool)
+            return {'success': True, 'detail': 'Image {!s} created successfully'.format(name)}
+        except (InvalidArgument, ImageExists) as e:
+            return {'success': False, 'detail': str(e)}
```

Comment:
1. You're returning here with a status code `201 Created`
2. just re-raising the exception would (at least for now) provide the same information to the frontend.

Reply:
>  (at least for now) 
Because, our `dashboard_v2.tools.json_error_page` handler is not getting the Exception. Instead, it's getting `str(e)` and the traceback. IMO, some interestingly bad API design. So, we should add an exception handler to `RESTController.default` instead, that knows what to do with Ceph exceptions.

Reply:
4. doing something like this:
```python
...
except (InvalidArgument, ImageExists) as e:
    return {'detail': str(e)}
```
is so generic, it shouldn't be implemented here.

Reply:
I don't think it's good to start generalize at the first time we see this code pattern. Let's wait for more code to be in to understand how we should refactor the error handling.